pyquil/pyqvm.py

```diff
@@ -199,26 +199,26 @@ def __init__(
                 quantum_simulator_type = ReferenceDensitySimulator
 
         self.n_qubits = n_qubits
-        self.ram = {}
+        self.ram: Dict[str, List[int]] = {}
 
         if post_gate_noise_probabilities is None:
             post_gate_noise_probabilities = {}
         self.post_gate_noise_probabilities = post_gate_noise_probabilities
 
-        self.program = None  # type: Program
-        self.program_counter = None  # type: int
-        self.defined_gates = dict()  # type: Dict[str, np.ndarray]
+        self.program: Optional[Program] = None
+        self.program_counter: Optional[int] = None
+        self.defined_gates: Dict[str, np.ndarray] = dict()
 
         # private implementation details
-        self._qubit_to_ram = None  # type: Dict[int, int]
-        self._ro_size = None  # type :int
-        self._memory_results = None  # type: Dict[str, np.ndarray]
+        self._qubit_to_ram: Optional[Dict[int, int]] = None
+        self._ro_size: Optional[int] = None
+        self._memory_results: Optional[Dict[str, np.ndarray]] = None
 
         self.rs = np.random.RandomState(seed=seed)
         self.wf_simulator = quantum_simulator_type(n_qubits=n_qubits, rs=self.rs)
         self._last_measure_program_loc = None
 
-    def load(self, executable):
+    def load(self, executable: Union[Program, PyQuilExecutableResponse]) -> "PyQVM":
```

Comment:
Is the quoted class name (`"PyQVM"`) for Python 3.6 compatibility, or does it serve another purpose?

Reply:
discussed offline